connexta/alliance
# CAL-36 Ingest MPEG-TS UDP Stream

#### What does this PR do?
Ingest MPEG-TS UDP Stream data.

#### Who is reviewing it?
@beyelerb @jrnorth @kcwire @rzwiefel 
#### How should this be tested?
Deploy the alliance-video-app to DDF. With the admin UI, go the Alliance Video App panel => Configuration => MPEG-TS UDP Stream Monitor. The default values are acceptable. Find the test code MpegTsUdpClient in alliance-mpegts-stream. Go to line 90 and adjust to the path to your local copy of 1003-GBR-IRL-SIT-H450-clip02-1003_OS.ts. (I know, this test app needs to be improved!) Every 30 seconds or so, the ddf log file should indicate that a new chunk is being stored. A ddf search should find the chunks and a parent metacard for the stream.
  
#### Any background context you want to provide?
I want to add some more unit tests, and to make sure the attributes temporal.start and temporal.end are being supplied by the MPEG-TS input transformer. But I wanted to make the code available for review because it is a large amount of code.

All of the files in libs/klv are just refactored code from alliance-mpegts-transformer. The refactoring occurred in CAL-29, but had to merged into this branch as well. 

#### What are the relevant tickets?

#### Screenshots (if appropriate)
#### Checklist:

Listen for UDP packets that contain MPEG-TS packets. Create a parent metacard for that
represents the stream. Break the data stream into chunks (based on elapsed time and
size) and store those chunks in the catalog. Each chunk's metacard is linked to the
parent metacard.
